Lib/fontTools/varLib/instancer.py

```diff
@@ -196,59 +200,108 @@ def _getVarRegionScalar(location, regionAxes):
 
 
 def _scaleVarDataDeltas(varData, regionScalars):
-    # multiply all varData deltas in-place by the corresponding region scalar
-    varRegionCount = len(varData.VarRegionIndex)
-    scalars = [regionScalars[regionIndex] for regionIndex in varData.VarRegionIndex]
+    # multiply all varData deltas in-place by the corresponding region scalar, and
+    # remove regions whose scalar is 0
```

Comment:
Won't there be a problem with indices if we drop the columns whose regionScalar == 0? I'm asking because the delta column gets dropped but the region still exists.  
If not can you add the explanation as a comment.

Reply:
The delta column gets dropped and the associated region is no longer referenced by this varData.VarRegionIndex, however the region stays in the VarRegionList until we call `VarStore.prune_regions()` at the end.
If we modified VarRegionList, all the VarRegionIndex would be invalidated.